File: site/cleanup_dropdown.py
file_path = r"c:\Users\HOME\Desktop\Mock Stream\site\IELTS Speaking Mocks.html"
with open(file_path, "r", encoding="utf-8") as f:
    lines = f.readlines()

new_lines = []

# Indices based on 1-based line numbers from view_file:
# Start skipping at Line 2943 (index 2942)
# Stop skipping at Line 3144 (index 3143)
start_skip_index = 2942
end_skip_index = 3143

for i, line in enumerate(lines):
    if i == start_skip_index:
        new_lines.append('          <option value="" disabled selected style="background:white;color:#667eea;font-weight:700;">Choose Mock\n')
        new_lines.append('          </option>\n')
        new_lines.append('          <option value="questions IELTS S/ielts-speaking-mock-1.js" style="background:white;color:#10b981;font-weight:600;">✓ Mock 01\n')
        new_lines.append('          </option>\n')
    
    if start_skip_index <= i <= end_skip_index:
        continue
        
    new_lines.append(line)

with open(file_path, "w", encoding="utf-8") as f:
    f.writelines(new_lines)

print("Modification complete.")

File: site/cleanup_dropdown.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'IELTS Speaking Mocks.html');
const content = fs.readFileSync(filePath, 'utf8');

// Split by newline, preserve type
const lines = content.split(/\r?\n/);
let outputLines = [];

for (let i = 0; i < lines.length; i++) {
    const lineIndex = i; // 0-based index corresponds to line number i+1

    // Choose Mock Option (Line 2943, 2944) -> Index 2942, 2943
    // Mock 01 Option (Line 2945, 2946) -> Index 2944, 2945
    // Mock 02 Option (Line 2947) -> Index 2946 ...

    // Replace Mock 01 line (Index 2944)
    if (lineIndex === 2944) {
        outputLines.push('          <option value="questions IELTS S/ielts-speaking-mock-1.js" style="background:white;color:#10b981;font-weight:600;">✓ Mock 01');
        continue;
    }

    // Remove from index 2946 up to index 3143 (Line 3144)
    // Line 3144 is the end of Mock 100 option.
    if (lineIndex >= 2946 && lineIndex <= 3143) {
        continue;
    }

    outputLines.push(lines[i]);
}

// Write back with consistent newlines (LF or CRLF based on platform, but here we just use what node defaults to or convert to LF)
// Using .join('\n') creates LF files usually.
fs.writeFileSync(filePath, outputLines.join('\n'), 'utf8');
console.log('Cleanup complete');
